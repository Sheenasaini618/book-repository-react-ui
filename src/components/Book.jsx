import * as React from 'react';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import ManualEditBook from './ManualEditBook';
import DeleteBook from './DeleteBook';


export default function Book(props) {

  var book = props.details
  console.log(book)
  return (
    <Card sx={{ maxWidth: 450, maxHeight: 500 }} style={{ marginTop: "10px" }}>
      <CardHeader
        avatar={
          <Avatar sx={{ bgcolor: red[500] }} aria-label="recipe">
            {book.title.charAt(0).toUpperCase()}
          </Avatar>
        }
        title={book.title}
        subheader={book.authors[0]}
      />
      <CardMedia
        component="img"
        sx={{ width: 151 }}
        height="194"
        image={book.imageLinks.smallThumbnail}
        alt="Paella dish"
      />
      <CardContent>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Price: â‚¹{book.price}.00
        </Typography>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Copies Available: {book.quantity}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <ManualEditBook details={book} />
        <DeleteBook details={book.id} />
      </CardActions>
    </Card>
  );
}
