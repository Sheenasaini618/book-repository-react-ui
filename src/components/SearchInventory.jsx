import * as React from 'react';
import Paper from '@mui/material/Paper';
import InputBase from '@mui/material/InputBase';
import Divider from '@mui/material/Divider';
import { retrieveBooksBasisTitle } from '../util/api/bookApi';
import BookCatalogue from './BookCatalogue';

export default function SearchInventory() {

    const [searchBookInventory, setsearchBookInventory] = React.useState("");
    const [data, setData] = React.useState([]);

    const handleChange = (e) => {
        setsearchBookInventory(
            e.target.value);
        console.log(e.target.value)
    };

    //componentDidMount and componentDidUpdate

    React.useEffect(() => {
        //Promise<Response>
        //Response
        console.log(searchBookInventory)
        retrieveBooksBasisTitle(searchBookInventory)
            .then((res) => {
                return res.json();
            })
            .then((result) => {
                setData(result);
            });
    }, [searchBookInventory]);

    return (
        <>
            <Paper
                component="form"
                sx={{ p: '2px 4px', display: 'flex', alignItems: 'center', width: 400 }}>
                <InputBase className='placeholder'
                    sx={{ ml: 1, flex: 1 }}
                    placeholder="Search Inventory book by title or author"
                    onChange={handleChange}
                />
                <Divider sx={{ height: 28, m: 0.5 }} orientation="vertical" />
            </Paper>
            <BookCatalogue details={data} />
        </>
    );
}
