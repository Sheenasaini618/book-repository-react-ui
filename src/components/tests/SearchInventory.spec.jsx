import React from 'react'
import Book from '../Book'
import '@testing-library/jest-dom'
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme'
import { configure } from 'enzyme';
import SearchInventory from '../SearchInventory'


configure({ adapter: new Adapter() });

let component

beforeEach(() => {
    component = shallow(<SearchInventory />);
});

describe('<SearchInventory />', () => {

    let wrapper;

    it('should validate placeholder in search Book Inventory component', () => {

        wrapper = shallow(<SearchInventory />);
        const event = {};
        expect(wrapper.find('.placeholder').length).toBe(1);
    })

    it('should match snapshot', () => {
        const element = component.debug()
        expect(element).toMatchSnapshot();
    });

})
