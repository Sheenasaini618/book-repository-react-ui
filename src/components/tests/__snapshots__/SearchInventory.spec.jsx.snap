// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInventory /> should match snapshot 1`] = `
"<Fragment>
  <ForwardRef(Paper) component=\\"form\\" sx={{...}}>
    <ForwardRef(InputBase) className=\\"placeholder\\" sx={{...}} placeholder=\\"Search Inventory book by title or author\\" onChange={[Function: handleChange]} />
    <ForwardRef(Divider) sx={{...}} orientation=\\"vertical\\" />
  </ForwardRef(Paper)>
  <BookCatalogue details={{...}} />
</Fragment>"
`;
